highfidelity/hifi
# Expose location scripting interface to Agent (AC script)

The location API will make Agent/AC scripts more location aware, and as a side effect they can also teleport to different domains.

## QA Test

 

1. Run the sandbox of this PR.
2. go to http://localhost:40100/settings/ and `Connect High Fidelity Account` to obtain a DomainID and new temporary place-name.
3. go to http://localhost:40100/assignment/ , and paste the following script:
    ```js
    var TIMEOUT_MS = 5000;

    Script.setTimeout(function() { 
        print('===========LOCATION INFO=============================');
        print('Location Info');
        print('Domain ID = ' + location.domainId);
        print('Placename = ' + location.placename);
        print('metaverseServerUrl = ' + location.metaverseServerUrl);
        print('isConnected = ' + (location.isConnected ? 'yes': 'no'));
        print('===========END LOCATION INFO=========================');
        Script.stop();
    }, TIMEOUT_MS);
    ```
4. Set number of instances to 1 , and press Run.
5. Now right-click on the the Server-Console a.k.a Sandbox in the traybar and click View Logs.
6. The Assignment-Clients tab should contain the `LOCATION INFO` about 5 seconds after you ran the script. It will look like:
    ```
    [06/21 07:24:11] [DEBUG] [hifi.scriptengine.script] [23728] [agent] ===========LOCATION INFO=============================
    [06/21 07:24:11] [DEBUG] [hifi.scriptengine.script] [23728] [agent] Location Info
    [06/21 07:24:11] [DEBUG] [hifi.scriptengine.script] [23728] [agent] Domain ID = {47dd212c-cab6-4711-a447-b04d6d425136}
    [06/21 07:24:11] [DEBUG] [hifi.scriptengine.script] [23728] [agent] Placename = azure-application-789
    [06/21 07:24:11] [DEBUG] [hifi.scriptengine.script] [23728] [agent] metaverseServerUrl = https://metaverse.highfidelity.com
    [06/21 07:24:11] [DEBUG] [hifi.scriptengine.script] [23728] [agent] isConnected = yes
    [06/21 07:24:11] [DEBUG] [hifi.scriptengine.script] [23728] [agent] ===========END LOCATION INFO=========================
    ```
7. Compare the obtained information with http://localhost:40100/settings/ , the place name should match, the domain ID should match, `metaverseServerUrl = https://metaverse.highfidelity.com` and `isConnected = yes`
